src/lib/client.ts

```diff
@@ -1,58 +1,66 @@
 import { providers } from "ethers";
 import {
   ERC20,
-  PerpDepository,
+  PerpDepository as PerpDepositoryContract,
   UXDController as UXDControllerContract,
 } from "../artifacts/types";
 import {
   ERC20__factory,
   PerpDepository__factory,
   UXDController__factory,
 } from "../artifacts/types/factories";
+import { PerpDepository } from "./PerpDepository";
 import { UXDController } from "./UXDController";
 
 export class UXDClient {
   protected readonly controller: UXDController;
+  protected readonly depository: PerpDepository;
 
   // -- TEMPORARY --
   // IN ORDER TO EXPOSE CONTRACTS WAITING FOR PROPER UTILITIES FUNCTIONS TO BE IMPLEMENTED
   protected readonly provider: providers.JsonRpcProvider;
   protected readonly redeemable: string;
   protected readonly controllerAddress: string;
-  protected readonly perpDepository: string;
+  protected readonly depositoryAddress: string;
   // ----------------
 
   constructor({
     provider,
     controller,
+    depository,
     redeemable,
-    market,
-    perpDepository,
   }: {
     provider: providers.JsonRpcProvider;
     controller: string;
+    depository: string;
     redeemable: string;
-    market: string;
-    perpDepository: string;
   }) {
     this.controller = new UXDController({
       provider,
       controller,
       redeemable,
-      market,
+    });
+
+    this.depository = new PerpDepository({
+      address: depository,
+      provider,
     });
 
     this.provider = provider;
     this.redeemable = redeemable;
     this.controllerAddress = controller;
-    this.perpDepository = perpDepository;
+    this.depositoryAddress = depository;
   }
 
   // TODO: more
   public getController(): UXDController {
     return this.controller;
   }
 
+  public getDepository(): PerpDepository {
```

Comment:
Why use getter functions rather than allowing direct instance property access ?

Could also have made the consumer instantiate controller/depository themselves and pass it to client constructor